nflow-tests/src/test/java/io/nflow/tests/BulkWorkflowTest.java

```diff
@@ -0,0 +1,120 @@
+package io.nflow.tests;
+
+import static io.nflow.engine.workflow.definition.BulkWorkflow.BULK_WORKFLOW_TYPE;
+import static io.nflow.engine.workflow.definition.BulkWorkflow.State.done;
+import static io.nflow.tests.demo.workflow.DemoBulkWorkflow.DEMO_BULK_WORKFLOW_TYPE;
+import static io.nflow.tests.demo.workflow.DemoWorkflow.DEMO_WORKFLOW_TYPE;
+import static java.util.Arrays.asList;
+import static java.util.Comparator.naturalOrder;
+import static java.util.stream.Collectors.toList;
+import static org.apache.cxf.jaxrs.client.WebClient.fromClient;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.joda.time.DateTime.now;
+import static org.junit.Assert.assertThat;
+import static org.junit.runners.MethodSorters.NAME_ASCENDING;
+
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import org.joda.time.DateTime;
+import org.junit.ClassRule;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.springframework.context.annotation.ComponentScan;
+
+import io.nflow.engine.workflow.definition.BulkWorkflow;
+import io.nflow.rest.v1.msg.CreateWorkflowInstanceRequest;
+import io.nflow.rest.v1.msg.CreateWorkflowInstanceResponse;
+import io.nflow.rest.v1.msg.ListWorkflowInstanceResponse;
+import io.nflow.rest.v1.msg.UpdateWorkflowInstanceRequest;
+import io.nflow.tests.demo.workflow.DemoBulkWorkflow;
+import io.nflow.tests.runner.NflowServerRule;
+
+@FixMethodOrder(NAME_ASCENDING)
+public class BulkWorkflowTest extends AbstractNflowTest {
+
+  @ClassRule
+  public static NflowServerRule server = new NflowServerRule.Builder().springContextClass(Configuration.class).build();
+
+  private static int workflowId;
+
+  private static int childrenCount;
+
+  public BulkWorkflowTest() {
+    super(server);
+  }
+
+  @ComponentScan(basePackageClasses = DemoBulkWorkflow.class)
+  static class Configuration {
+    // for component scanning only
+  }
+
+  @Test
+  public void t01_startDemoBulkWorkflow() {
+    CreateWorkflowInstanceRequest req = new CreateWorkflowInstanceRequest();
+    req.type = DEMO_BULK_WORKFLOW_TYPE;
+    req.stateVariables.put(BulkWorkflow.VAR_CONCURRENCY, 3);
+    List<Integer> childData = asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
```

Comment:
Replace `asList` with `IntStream.range(1, CHILD_COUNT).boxed().collect(toList())`

Reply:
If wanting to express the real count of the children with the CHILD_COUNT constant, rather use `IntStream.rangeClosed(1, CHILD_COUNT).boxed().collect(toList())`

Reply:
Fixed